src/addons/addonmessage.cpp

```diff
@@ -74,35 +74,97 @@ AddonMessage::AddonMessage(QObject* parent, const QString& manifestFileName,
 
 AddonMessage::~AddonMessage() { MVPN_COUNT_DTOR(AddonMessage); }
 
-void AddonMessage::dismiss() {
-  m_dismissed = true;
-  disable();
-
+// static
+AddonMessage::State AddonMessage::loadMessageState(const QString& id) {
   SettingsHolder* settingsHolder = SettingsHolder::instance();
-  Q_ASSERT(settingsHolder);
 
-  QStringList dismissedAddonMessages = settingsHolder->dismissedAddonMessages();
-  dismissedAddonMessages.append(id());
-  settingsHolder->setDismissedAddonMessages(dismissedAddonMessages);
-}
+  QStringList notifiedAddonMessages = settingsHolder->notifiedAddonMessages();
+  if (notifiedAddonMessages.contains(id)) {
+    return State::Notified;
+  }
 
-void AddonMessage::maskAsRead() {
-  if (m_isRead) {
-    return;
+  QStringList readAddonMessages = settingsHolder->readAddonMessages();
+  if (readAddonMessages.contains(id)) {
+    return State::Read;
   }
 
-  m_isRead = true;
+  QStringList dismissedAddonMessages = settingsHolder->dismissedAddonMessages();
```

Comment:
This can introduce bugs where, for funny reasons, we have the same ID in the dismiss setting and in the read setting...and so on. What about if we change this all and we have 1 setting only: `addonMessageStates` (or something better).

I was wondering if it makes sense to do this:
```
SettingsHolder* settingsHolder = SettingsHolder::instance();
Q_ASSERT(settingsHolder);

AddonMessage::State state = settingsHolder->addonProperty(addon->id(), "state")->castingToStateSomehow();
...
```

Basically we can add in the SettingsHolder a group for each addon: https://doc.qt.io/qt-5/qsettings.html#beginGroup
The group can be called `addons/<id>`.

Reply:
Addressed in [1320edd](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/4327/commits/1320edd8c83262044dcc0837c16504d5aa3158ac)